### PR TITLE
sby: refactor

### PR DESCRIPTION
**Pull request by RCoeurjoly**
## Description of changes
Fix description. Remove duplicated inputs. --replace-fail instead of --replace. Be specific about which files patchShebangs is applied to. Run hooks in installPhase. Use btor2tools as input now that is updated

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by pbsds — APPROVED** [review]
Thank you for fixing this up, diff LGTM, and a nixpkgs-review is on the way. I have a plane to catch now

Please shorten the title, a simple `sby: refactor` is okay. Details go in the commit body

File: pkgs/by-name/sb/sby/package.nix [hunk 1]
```diff
@@ -36,49 +37,60 @@ stdenv.mkDerivation rec {
...
+    patchShebangs tests/autotune/autotune_div.sh
+    patchShebangs tests/autotune/autotune_options.sh
+    patchShebangs tests/keepgoing/keepgoing_same_step.sh
+    patchShebangs tests/keepgoing/keepgoing_smtc.sh
+    patchShebangs tests/keepgoing/keepgoing_multi_step.sh
+    patchShebangs tests/junit/junit_assert.sh
+    patchShebangs tests/junit/junit_timeout_error.sh
+    patchShebangs tests/junit/junit_expect.sh
+    patchShebangs tests/junit/junit_cover.sh
+    patchShebangs tests/junit/junit_nocodeloc.sh
```
Comment (SuperSandro2000) [inline pkgs/by-name/sb/sby/package.nix:55]:
```suggestion
    patchShebangs docs/source/conf.py \
      docs/source/conf.diff \
      tests/autotune/*.sh \
      tests/keepgoing/*.sh \
      tests/junit/*.sh
```

**Comment by thoughtpolice** [comment]
Let's squash these two commits (GitHub creates a commit for "Suggested changes" which is annoying) and it LGTM, I'll merge after.

**Comment by RCoeurjoly** [comment]
> Let's squash these two commits (GitHub creates a commit for "Suggested changes" which is annoying) and it LGTM, I'll merge after.

Done. Thanks for the suggestion

**Review by thoughtpolice — APPROVED** [review]

**Pull request merged**